src/vs/workbench/contrib/scm/browser/dirtydiffDecorator.ts

```diff
@@ -1146,16 +1131,23 @@ export class DirtyDiffModel extends Disposable {
 
 		this._register(this._model.onDidChangeEncoding(() => {
 			this.diffDelayer.cancel();
-			this._originalResource = null;
-			this._originalModel = null;
-			this._originalURIPromise = undefined;
+			this._quickDiffs = null;
+			this._originalModels.clear();
+			this._quickDiffsPromise = undefined;
 			this.setChanges([]);
 			this.triggerDiff();
 		}));
 
 		this.triggerDiff();
 	}
 
+	public diffEditorModel(originalUri: string): IDiffEditorModel {
+		return {
+			modified: this._model.textEditorModel!,
+			original: this._originalModels.get(originalUri)?.textEditorModel!
```

Comment:
Isn't this risky? What if `originalUri` isn't in `this._originalModels`?

Reply:
Yes, definitely risky.